crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -268,6 +271,21 @@ where
         // Step 0: Pre-processing
         self.semantic.push_node(stmt);
 
+        // For Jupyter Notebooks, we'll reset the `IMPORT_BOUNDARY` flag when
+        // we encounter a cell boundary.
+        if self.settings.preview.is_enabled()
+            && self.source_type.is_ipynb()
+            && self.semantic.at_top_level()
+            && self.semantic.seen_import_boundary()
+            && self.cell_offsets.is_some_and(|cell_offsets| {
+                cell_offsets.has_cell_boundary(TextRange::new(self.last_stmt_end, stmt.start()))
+            })
+        {
+            self.semantic
+                .flags
+                .remove(SemanticModelFlags::IMPORT_BOUNDARY);
```

Comment:
Is `-=` equivalent here? It looks like we use that elsewhere in the file.

Reply:
Oh, I didn't notice that. Although they're equivalent, I'll change it to `-=` for consistency.